app/controllers/events_controller.rb

```diff
@@ -0,0 +1,75 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:show, :edit, :update, :destroy]
+  before_action :authorize_project_member
+  before_action :authorize_event_author, only: [:update, :destroy]
+
+  def index
+    @project = Project.find(params[:project_id])
```

Comment:
it does but then I would need to pass 2 variables to my view, I don't think that it's mixing concerns if I just pass `Project` model to my view. The controller uses both `Event` and `Project` models because it's nested as project's resource.